### PR TITLE
bump go version

### PR DESCRIPTION
**Pull request by maili-cb**
Fixes # .

### Motivation
Upgrade Go to 1.16

### Solution
Upgrade Go to 1.16

### Open questions

Sister PR: https://github.com/coinbase/rosetta-ethereum/pull/58

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -2,12 +2,16 @@
...
+ADDLICENSE_CMD=addlicense
 ADDLICENCE_SCRIPT=${ADDLICENSE_CMD} -c "Coinbase, Inc." -l "apache" -v
 SPELLCHECK_CMD=go run github.com/client9/misspell/cmd/misspell
-GOLINES_CMD=go run github.com/segmentio/golines
-GOLINT_CMD=go run golang.org/x/lint/golint
-GOVERALLS_CMD=go run github.com/mattn/goveralls
+GOLINES_INSTALL=go install github.com/segmentio/golines@latest
+GOLINES_CMD=golines
+GOLINT_INSTALL=go install golang.org/x/lint/golint
+GOLINT_CMD=golint
```
Comment (maili-cb) [inline Makefile:12]: cause failure in `make lint`

File: go.mod [hunk 2]
```diff
@@ -10,6 +10,7 @@ require (
 	github.com/grpc-ecosystem/go-grpc-middleware v1.2.2
 	github.com/stretchr/testify v1.6.1
 	go.uber.org/zap v1.16.0
+	golang.org/x/lint v0.0.0-20210508222113-6edffad5e616 // indirect
```
Comment (shrimalmadhur) [inline go.mod:13]: This should not be here IMO since this is only used while building. `go mod tidy` should have removed this?

**Review by shrimalmadhur — APPROVED** [review]
LGTM. Thanks!

**Review by shrimalmadhur — APPROVED** [review]

**Review by shrimalmadhur — APPROVED** [review]

**Review by rthille — APPROVED** [review]

**Comment by cb-heimdall** [comment]
Review Error for rthille @ 2021-09-16 19:21:11 UTC
User must have write permissions to review

**Review by shrimalmadhur — APPROVED** [review]

**Pull request merged**